File: vm/home/tango-cs/training/client-api/writing01.cpp
/*
 * writng01.cpp
 * 
 *  example program for Tango training - writing an attribute
 * 
 * C. Scafuri
 * Elettra - Sincrotrone Trieste S.C.p.A.
 * 2016
 */



#include <tango.h>
#include <cstdlib>
using namespace Tango;
int main(int argc, char **argv)
{
    string devname = "sys/tg_test/1";
    string attname = "double_scalar";
    DeviceProxy *dev = 0;
    // step 1) create the device proxy
    try{
        dev = new Tango::DeviceProxy(devname);
    }
    catch(Tango::DevFailed &e)
    {
        Tango::Except::print_exception(e);
        cerr << "failed to create DeviceProxy" << endl;
        exit(EXIT_FAILURE); // trivial error handling

    }
    
    // prepare the
    double value=1.4142;
    DeviceAttribute da(attname,value); //use direct constructor for deviceAttribute
    
    // DeviceAttribute da;
    // da.set_name("double_scalar");
    // double dblvalue = 5.678;
    // da << dblvalue;   //or use insertion operator!

    try{
        dev->write_attribute(da); //write the attribute
    }
    catch(DevFailed &e)
    {
        Tango::Except::print_exception(e);
        cerr << "failed to write attribute" << endl;
        delete dev; // free resources
        exit(EXIT_FAILURE); // trivial error handling
    }
    delete dev;
    exit(EXIT_SUCCESS);
}

File: vm/home/tango-cs/training/python-api/writing01.py
#/!/usr/bin/python
#
# client example: wrting an attribute , also with simplified syntax
# C. Scafuri - jan. 2016
#

from PyTango import *
import sys

devname = 'sys/tg_test/1'
attname = 'double_scalar'

#step 1) create the device proxy
try:
    dev = DeviceProxy(devname)
    print 'proxy for %s created' %(devname)
except DevFailed, ex:
        Except.print_exception(ex)
        sys.exit(1) #semplicistic error handling
try:
    dev.write_attribute(attname,5.678) #write the attribute directly
    print 'write_attribute "%s" OK' % (attname)
except DevFailed, ex:
    Except.print_exception(ex)
    print 'failed to write attribute'
    sys.exit(1) #simplicistic error handling
    
# use simplified syntax:

try:
    dev.double_scalar=0.619
    print 'write_attribute SIMPLIFIED "%s" OK' % (attname)
except DevFailed, ex:
    Except.print_exception(ex)
    print 'failed to write attribute with simplified sytax'
    sys.exit(1) #simplicistic error handling

sys.exit(0)
